File: app/src/main/java/com/zibete/proyecto1/Adapters/SliderPhotoAdapter.java
package com.zibete.proyecto1.Adapters;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import com.bumptech.glide.Glide;
import com.github.chrisbanes.photoview.PhotoView;
import com.zibete.proyecto1.R;

import java.util.ArrayList;

public class SliderPhotoAdapter extends PagerAdapter {

    private Context context;
    private LayoutInflater layoutInflater;
    ArrayList <String> photoList;
    PhotoView photo;
    int rotation;


    public SliderPhotoAdapter(Context context, ArrayList<String> photoList, int rotation){
        this.context = context;
        this.photoList = photoList;
        this.rotation = rotation;
    }


    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {

        layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = layoutInflater.inflate(R.layout.slide_photo_layout, container, false);

        photo = view.findViewById(R.id.photofinal);
        photo.setRotationY(rotation);

        Glide.with(context).load(photoList.get(position)).into(photo);

        photo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((Activity) context).finish();
            }
        });



        container.addView(view);
        return view;
    }


    @Override
    public int getCount() {
        return photoList.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == (LinearLayout) object;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((LinearLayout)object);
    }


}

File: app/src/main/java/com/zibete/proyecto1/adapters/SliderPhotoAdapter.kt
package com.zibete.proyecto1.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import coil.load
import coil.size.Scale
import com.zibete.proyecto1.databinding.ItemSlidePhotoBinding

class SliderPhotoAdapter(
    private val urls: List<String>,
    private val onLoadStart: (position: Int) -> Unit,
    private val onLoadEnd: (position: Int) -> Unit
) : RecyclerView.Adapter<SliderPhotoAdapter.VH>() {

    inner class VH(val binding: ItemSlidePhotoBinding) : RecyclerView.ViewHolder(binding.root) {
        val photoView get() = binding.photoView
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val binding =
            ItemSlidePhotoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return VH(binding)
    }

    override fun getItemCount(): Int = urls.size

    override fun onBindViewHolder(holder: VH, position: Int) {
        val url = urls.getOrNull(position).orEmpty()

        holder.photoView.load(url) {
            crossfade(true)
            scale(Scale.FIT)

            listener(
                onStart = { onLoadStart(position) },
                onSuccess = { _, _ -> onLoadEnd(position) },
                onError = { _, _ -> onLoadEnd(position) },
                onCancel = { onLoadEnd(position) }
            )
        }
    }
}

